movemap.h

```diff
@@ -8,20 +8,31 @@
 #include "element.h"
 
 /**
- * @brief 
+ * Contains information about a sequence of moves in a convenient format
+ * used in the move function of Board.
+ * 
+ * Holds two maps, one for inputs of the move and one for the outputs.
+ * Each map has keys for the elements the spheres the gems are moving from/to,
+ * and values containing map of each gem type and the number of those gems involved
+ * in the move.
  * 
  */
 class MoveMap {
   public:
-  MoveMap(const std::vector<std::tuple<Element, Element, Element>>);
-  int inputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
-  int outputCount(const int sphereType = Element::ALL, const int gemType = Element::ALL) const;
+    // takes the moves in the form of a vector containing a tuple of moves in 
+    // {gem type, source, destination}.
```

Comment:
I don't see a reason to comment this. The order is useful, but you could add that to the class header, considering it's used all over the place. Other than that, I think the code is clear enough.